pkg/provisioning/config/enrich.go

```diff
@@ -23,11 +25,33 @@ func Enrich(cfg Pipeline) Pipeline {
 		cfg.Status = StatusRunning
 	}
 
+	cfg.DLQ = enrichDLQ(cfg.DLQ)
 	cfg.Connectors = enrichConnectors(cfg.Connectors, cfg.ID)
 	cfg.Processors = enrichProcessors(cfg.Processors, cfg.ID)
 	return cfg
 }
 
+func enrichDLQ(dlq DLQ) DLQ {
+	if dlq.Plugin == "" {
+		dlq.Plugin = pipeline.DefaultDLQ.Plugin
+	}
+	if dlq.Settings == nil {
+		dlq.Settings = make(map[string]string, len(pipeline.DefaultDLQ.Settings))
+		for k, v := range pipeline.DefaultDLQ.Settings {
+			dlq.Settings[k] = v
+		}
+	}
+	if dlq.WindowSize == nil {
+		tmp := pipeline.DefaultDLQ.WindowSize
+		dlq.WindowSize = &tmp
+	}
+	if dlq.WindowNackThreshold == nil {
+		tmp := pipeline.DefaultDLQ.WindowNackThreshold
+		dlq.WindowNackThreshold = &tmp
+	}
+	return dlq
```

Comment:
Is there a special intention behind mutating the input value `dlq` and also returning it?

Reply:
The mutations are not visible to the caller, `DLQ` is passed in as a value, so we have to return it.